### PR TITLE
create a single kms key for encryption inside ec2 module

### PR DESCRIPTION
**Pull request by ndegwajohn**
# Description

This PR does the following:
 - modifies the module code to create only one kms key for encryption.
 - adds a `kms_key_id` variable to provide an external kms key.
 - adds new GitHub workflow and config files.
 - fixes CKV_AWS_149 and CKV_AWS_158 .

## Features/Fixes/Patches/Changes list:

Please check the [CHANGELOG.md](https://github.com/boldlink/terraform-aws-ec2/blob/feature/create-single-key/CHANGELOG.md#120---2023-01-10)

### PR Owners Checklist:
* [x] Have you followed the guidelines in our Contributing document?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../../pulls) for the same update/change?
* [x] Have you checked there aren't open [Pull Requests](../../../pulls) of upstream or parallel code that will cause conflicts?
* [x] Have you updated the `CHANGELOG.md`?
* [x] Have you updated the `README.md`(s)?
* [x] OWNER: Does your submission pass?
    * [x] Pre-commit (attach logs/print screen to this PR)
    * [x] Checkov/terrascan (attach logs/print screen to this PR)
    * [x] Terraform resources examples build/destroy successfully with Terraform 0.14.11 minimum?
* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [x] Have you written new tests for your core changes, as applicable?
* [x] Have you successfully run tests with your changes locally?

### PR Reviewers Checklists?
* [ ] Are all your comments/changes resolved?
* [ ] Are you happy with the OWNER checklists and additional information provided?
* [ ] Does your review tests pass?
    * [ ] Terraform resources examples build/destroy successfully with Terraform 0.14.11 minimum?

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,13 +1,19 @@
...
 [<img src="https://avatars.githubusercontent.com/u/25388280?s=200&v=4" width="96"/>](https://boldlink.io)
 
 # AWS EC2 Terraform module
 
 ## Description
 
-This terraform module creates an EC2 Instance with a security group, Cloudwatch Log Group for EC2 detailed monitoring, KMS key to encrypt/decrypt the ebs volumes, and another KMS key for Log Group. Keypair and Instance IAM role creation is optional.
+This terraform module creates an EC2 Instance with a security group, Cloudwatch Log Group for EC2 detailed monitoring, a KMS key to encryption, Keypair and Instance IAM role creation is optional.
 
 Examples available [here](https://github.com/boldlink/terraform-aws-ec2/tree/main/examples/)
```
Comment (patrickmukumbu) [inline README.md:19]: This was an overlooked error in previous releases. You can correct this by typing the followin
`Examples available [here](./examples/)

File: README.md [hunk 2]
```diff
@@ -1,13 +1,19 @@
...
+[![Build Status](https://github.com/boldlink/terraform-aws-ec2/actions/workflows/auto-badge.yaml/badge.svg)](https://github.com/boldlink/terraform-aws-ec2/actions)
 
 [<img src="https://avatars.githubusercontent.com/u/25388280?s=200&v=4" width="96"/>](https://boldlink.io)
 
 # AWS EC2 Terraform module
 
 ## Description
 
-This terraform module creates an EC2 Instance with a security group, Cloudwatch Log Group for EC2 detailed monitoring, KMS key to encrypt/decrypt the ebs volumes, and another KMS key for Log Group. Keypair and Instance IAM role creation is optional.
+This terraform module creates an EC2 Instance with a security group, Cloudwatch Log Group for EC2 detailed monitoring, a KMS key to encryption, Keypair and Instance IAM role creation is optional.
```
Comment (patrickmukumbu) [inline README.md:17]: As advised earlier by @hugoalmeida264  please highlight the benefits of using this module over other modules for the same AWS service.

**Review by patrickmukumbu — CHANGES_REQUESTED** [review]
See my comments @ndegwajohn

File: README.md (hunk 1 above)
Comment (ndegwajohn) [reply README.md:19]: Alright

**Review by patrickmukumbu — APPROVED** [review]
LGTM Now

**Review by patrickmukumbu — APPROVED** [review]

**Review by patrickmukumbu — APPROVED** [review]

**Pull request merged**